zip-0227.rst

```diff
@@ -1,8 +1,359 @@
+
+The protocol was designed with three main properties in mind:
+- at the time of issuance, the issuer can already allocate all the tokens to the corresponding owners by creating the corresponding (shielded) output notes to the respective addresses. As is implied, the issuance mechanism is itself transparent, but the allocation is totally private
+- in a single issuance bundle, the issuer can publish many issuance actions, or in other words, create multiple custom assets
+- every issuance action contains the `finiteIssuance` boolean that defines whether that specific custom asset can have further tokens issued or not
+
+The issuance mechanism is broad enough for issuers to either create native assets on Zcash (i.e.: assets that originate on the Zcash blockchain), as well as for institutions to create bridges from other chains and import wrapped assets. In fact, the protocol described below enables, what we hope will be a useful set of, applications. The design decision is not to have a chosen name to describe the custom asset, but to delegate it to an off-chain mapping, as this would imply a land-grab “war”.
+
+Issuance Keys
+-------------
+
+There are two keys relevant to the Issuer:
+
+1. The `issuanceAuthorizingKey`, denoted as `isk`, is the key that is used to sign the issuance transaction. This key is used to authorize the issuance of a specific asset type, and is only used by the issuer. Throughout we also call this key the issuer signature key.
+
+2. The `issuanceValidatingKey`, denoted as `ik`, is the key that is used to validate the issuance transaction. This key is used to validate the issuance of a specific asset type, and is used by all blockchain users (specifically the asset owners and consensus validators) to associate the asset in question with the issuer. Throughout we also call this key the issuer verification key.
+
+The two keys are derived in an analogous manner to the `spendAuthorizingKey` and `spendValidatingKey` keys, using the same signature scheme, as described in ZIP 32 [#zip-0032]_ and in accordance with ZIP 316 [#zip-0316]_.
+
+- The `issuanceAuthorizingKey` is derived directly from the `spendingKey`, `sk`, as a private signature key:
+
+:math:`\mathsf{isk := ToScalar^{Orchard}(︀ PRF^{expand}_{sk} ([0x0a])}`
+
+- The `issuanceValidatingKey` is derived from the `issuanceAuthorizingKey`, `isk`, as a public verification key:
+
+:math:`\mathsf{ik := SpendAuthSig^{Orchard}.DerivePublic(isk)}`
+
+This allows the issuer to use the same wallet it usually uses to transfer assets, while keeping a disconnect from the other keys. Specifically, this method is aligned with the requirements and motivation of ZIP 32 [#zip-0032]_, and can further be in  It provides further anonymity and the ability to delegate issuance of an asset (or in the future, generate a multi-signature protocol) while the rest of the keys remain in the wallet safe.
```

Comment:
It's unclear what "the `spendingKey`" refers to. If this is the Orchard spending key for a ZIP32 account, we think this is undesirable because the issuance protocol might need to be used for multiple shielded pools.

It is not necessary to tie the issuance key to a particular Orchard spending key in order for the issuance key to be recoverable from a wallet seed. We should instead carve out a namespace within the ZIP 32 hierarchy so that issuance keys can be derived from a wallet's seed phrase without being dependent on a particular protocol's key tree.

Reply:
Yes, we were referring to the Orchard spending key. Is the undesirability due to having to revert back to Orchard key generation for issuance even after moving to a future shielded pool, for example? Are there other reasons as well?

> We should instead carve out a namespace within the ZIP 32 hierarchy so that issuance keys can be derived from a wallet's seed phrase without being dependent on a particular protocol's key tree

Issuance keys could instead be derived from a key generated from a wallet seed like in [Orchard master key generation](https://zips.z.cash/zip-0032#orchard-master-key-generation) with a new personalisation (eg. `ZcashIP32Issuance`).

Reply:
We have recently made these changes, moving to a separately generated issuance key structure that gets generated from the wallet seed.